src/server.c

```diff
@@ -132,8 +133,14 @@ void serverLogRaw(int level, const char *msg) {
         } else {
             role_char = (server.masterhost ? 'S':'M'); /* Slave or Master. */
         }
-        fprintf(fp,"%d:%c %s %c %s\n",
-            (int)getpid(),role_char, buf,c[level],msg);
+        const char *pre = "";
+        const char *post = "";
+        if (isatty(fileno(fp))) {
+            pre = level_color[level];
+            post = "\033[0m";
+        }
+        fprintf(fp,"%d:%c %s %s%c%s %s\n",
+            (int)getpid(),role_char, buf, pre, c[level], post, msg);
```

Comment:
any reason not to move the `post` to after `msg`?

Reply:
I wanted to keep the messages in the default console color to make it easier on the eye. In other applications I tend two see both approaches (coloring the level prefix or coloring the whole log line).